### PR TITLE
Add nanobind_bazel v2.4.0

### PR DESCRIPTION
**Pull request by nicholasjng**
Mirrors nanobind v2.4.0. Note that no nanobind v2.3.0 was released due to a mistake in the release process.

### CODE REVIEW

**Comment by nicholasjng** [comment]
I guess Bazel 6.x can go from the presubmit, but Bazel 7.x now also fails (missing `--std="c++17"`), which is confusing to me.

**Comment by nicholasjng** [comment]
Decided to set the test target in the presubmit to the newly exposed `@nanobind_bazel//:nanobind`, which was the "headline feature" for v2.4.0.

(For clarification: This is an alias pointing to the `@nanobind` repo extension - it was there since the beginning, because it's the whole point of nanobind_bazel, but it was not accessible by downstream users until this release.)

**Review by fmeum — DISMISSED** [review]

**Comment by nicholasjng** [comment]
Do you have an idea on how to inject C++17 as a standard into the Bazel 7.x job? I don't have it in my copts / cxxopts explicitly.

File: modules/nanobind_bazel/2.4.0/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    - macos_arm64
+    - windows
+
+tasks:
+  verify_targets:
+    name: Verify build targets
+    platform: ${{ platform }}
+    bazel: ${{ bazel }}
+    build_targets:
+    - '@nanobind_bazel//:nanobind'
```
Comment (fmeum) [inline modules/nanobind_bazel/2.4.0/presubmit.yml:18]:
```suggestion
    build_targets:
    - '@nanobind_bazel//:nanobind'
    build_flags:
    - --cxxopt=-std=c++17
```

Comment (nicholasjng) [reply modules/nanobind_bazel/2.4.0/presubmit.yml:18]: Are those cross-platform? Then I would also have to put `--cxxopt=/std:c++17` for MSVC/Windows most likely.

Comment (nicholasjng) [reply modules/nanobind_bazel/2.4.0/presubmit.yml:27]: Resolved by unrolling the platform and jobs.

**Review by fmeum — APPROVED** [review]

**Pull request merged**